### PR TITLE
changed all longitudinal functions to arrow functions

### PR DESCRIPTION
**Pull request by jpa399**
_Pull requests into Flux require the following. Submitter should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable (**N/A**) and :white_check_mark:. Reviewers should complete every item on the bulleted list when reviewing._

Changed all non-lifecycle functions in LongitudinalTable and LongitudinalTableVisualizer to arrow functions, so they now align with the Flux Notes style guide.  Functionality of the table was not affected.
## Submitter:

**Running the Application**

- [x] Manually tested in Chrome (primary browser for testing)
- [ ] Manually tested in IE/Safari (verify app loads and basic feature testing)
- [x] Manually tested in Chrome in iPad mode (at minimum, viewing patient record)

**Task**

- [x] Branch implements task as expected
- [x] Key application features are updated as needed for task and tested (see: [Key Application Features](https://github.com/FluxNotes/flux/wiki/Key-Application-Features))
- [x] Task specific definition of done (as described in the Jira task) is met, if applicable.

**Documentation**

- [x] This pull request describes why these changes were made
- [x] Recognizes any potential shortcomings/bugs in the description 
- [ ] Pre release script is updated on the [Pre Release Testing Script wiki page](https://github.com/FluxNotes/flux/wiki/Pre-Release-Testing-Script) (add/remove single features to test, do not run through full script)
- [ ] Update the list of key features with new features on the [Key Application Features wiki page](https://github.com/FluxNotes/flux/wiki/Key-Application-Features)
- [ ] Document any new APIs/extension points
- [ ] Additional technical components and libraries are documented and added to [wiki](https://github.com/FluxNotes/flux/wiki/About-Flux-Notes) as needed

**Code Quality**

- [x] Code style guide is followed (see: [Code Style Guide](https://github.com/FluxNotes/flux/wiki/JavaScript-and-HTML-Code-Style-Guide))
- [x] Code is commented

**Tests**

- [x] Existing tests passed
- [ ] Added backend tests for new functionality added

## Reviewers:

- Check that code is maintainable and reusable. Code reuses existing code and infrastructure where appropriate.
- Check the PR and code accomplishes the task's purpose.
- Check that new tests appropriately test the new code, including edge cases. Other existing tests pass.
- Verify Pull Request checklist is correctly completed by submitter.
- Try to break the code

### CODE REVIEW

File: src/summary/LongitudinalTableVisualizer.jsx [hunk 1]
```diff
@@ -66,13 +66,13 @@ export default class LongitudinalTableVisualizer extends Visualizer {
         const finalArray = _.concat(favsArray,notFavsArray); //but the favs back together with the not favs in one array
         this.setState({ data: finalArray });
     }
-    componentWillReceiveProps(nextProps) {
+    componentWillReceiveProps = (nextProps) => {
```
Comment (jafeltra) [inline src/summary/LongitudinalTableVisualizer.jsx]: [JA1]: I think since we don't want to change the React lifecycle methods, this shouldn't be an arrow function.

**Comment by jafeltra** [comment]
This looks good! I just added one minor comment!

File: src/summary/LongitudinalTableVisualizer.jsx (hunk 1 above)
Comment (jpa399) [reply src/summary/LongitudinalTableVisualizer.jsx]: you're right, I must've changed that one by accident.  Fixed it now, thanks!

**Comment by jafeltra** [comment]
Looks good! I've already approved and will merge once Travis passes.
